### PR TITLE
FEATURE: support skipping unpublishing of resources

### PR DESCRIPTION
**Pull request by skurfuerst**

### CODE REVIEW

File: Classes/S3Target.php [hunk 1]
```diff
@@ -234,12 +245,16 @@ public function publishCollection(CollectionInterface $collection, callable $cal
...
-            ));
+        if ($this->unpublishResources !== false) {
+            foreach (array_keys($potentiallyObsoleteObjects) as $relativePathAndFilename) {
+                $this->systemLogger->debug(sprintf('Deleted obsolete resource "%s" from bucket "%s"', $relativePathAndFilename, $this->bucketName));
+                $this->s3Client->deleteObject(array(
+                    'Bucket' => $this->bucketName,
+                    'Key' => $this->keyPrefix . $relativePathAndFilename
+                ));
+            }
+        } else {
```
Comment (kitsunet) [inline Classes/S3Target.php]: Could turn this around and do just logging and early return in the IF part and then do the unpublishing after outside of the conditionna.

**Comment by kitsunet** [comment]
Probably makes sense for some applications.

File: Classes/S3Target.php (hunk 1 above)
Comment (skurfuerst) [reply Classes/S3Target.php]: will do :) :+1:

Comment (skurfuerst) [reply Classes/S3Target.php]: will do :) :+1:

**Comment by batabana** [comment]
Unfortunately I didn't see this PR before so I kind of added a new topic on top of it - sorry for the mess. 

So to clear things up: I saw the review comment for the first feature (support skipping unpublishing of resources) just now and restructured the code accordingly.
There was a new feature added before: Support disabling the standard ACL setting of the target (`public-read`) which is needed in some cases, e.g. when using cloudfront with conflicting restrictive policies.

**Review by kitsunet — APPROVED** [review]

**Pull request merged**